### ISSUE TITLE
Move the `entries` from the `file` object into the version object for Browse and Compare APIs

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/addons-server#14212]
The `entries`, which is the tree of all files that are included in an uploaded add-on, are currently returned in the `file` object in the API responses for Browse and Compare. When the APIs are called with the `file_only` param, these `entries` are empty. This design isn't ideal, because these `entries` really apply to the `version` and not the `file`, because the `file` now represents the requested file, not the xpi file. This also makes things more difficult for types in code manager.

To resolve this, and have the data where it really belongs, we should move these `entries` from the `file` object up to the root of the response, which is where the other `version` properties are. While we're at it, it probably makes sense to rename the property from `entries`. I was thinking `file_entries`, `filetree` or maybe even just `files`. I think the latter makes sense, but it might be a bit confusing to have both a `file` property and a `files` property. If anyone has a suggestion for what to call this property once it has been moved from `file` to `version`, please let me know.

**Comment by bobsilverberg** [comment]
@willdurand FYI, changing this will help a bit with the TS issues I was having when trying to update code manager to use the new `file_only` APIs. Getting rid of an optional property, which really doesn't belong there at all, makes things simpler for defining types as well.

**Comment by ioanarusiczki** [comment]
@bobsilverberg I made some API requests on AMO dev - Browse&Compare
- `file_entries` have been moved outside `file` and the information is returned into the response when `file_only` parameter is not used (or set to false). 
- the `file_only`:true will only show file information for a specific file when also `file` parameter is used into the API request. Without the `file` parameter the default is manifest.json.
